connections/multiselect2.md

```diff
@@ -0,0 +1,108 @@
+# Multiselect 2.0
+
+## Introduction
+
+Multiselect 2.0 replaces the Multistream protocol. Compared to its predecessor, it offers:
+
+1. Downgrade protection for the security protocol negotiation.
+2. Zero-rountrip stream multiplexer negotiation for handshake protocols that take advantage of early data mechanisms (one-roundtrip negotation for protocols / implementations that don't).
+3. Compression for the protocol identifiers of frequently used protocols.
+
+By using protobufs for all control messages, Multiselect 2.0 provides an easy path for future protocol upgrades. The protobuf format guarantees that unknown field in a message will be skipped, thus future version of the protocol can add new fields that signal support for new protocol features.
+
+## High-Level Overview
+
+### Secure Channel Selection
+
+Conversely to multistream-select 1.0, secure channel protocols are not dynamically negotiated in multiselect 2.0. Instead, they are announced upfront in the peer multiaddrs. It is up to the implementers to decide whether each secure channel is exposed over a different port, or if a single port handles all secure channels, and a demultiplexing strategy is used to identify which protocol is being used.
+
+**TODO**: Do we need to describe the format here? I guess we don't, but we will probably need another document for that change, and we can link to it from here.
+
+Peers advertising a multiaddr that includes a handshake protocol MUST support Multiselect 2.0 as described in this document.
+
+#### TCP Simultaneous Open
```

Comment:
From the perspective of this spec, we should not talk about specific transports, but rather about "role deduplication" or something like that. The TCP simultaneous open is a specific instance where this deduplication is necessary, and at the connection bootstrapping layer, the transport mechanics are irrelevant.

Reply:
@tomaka already raised that point in https://github.com/libp2p/specs/pull/227#discussion_r350673105, and we agreed to move the actual logic to the transport / secure channel specification, while requiring these to define *some way* to deal with TCP sim open in this document.